drivers/vulkan/rendering_device_vulkan.cpp

```diff
@@ -1963,7 +1963,7 @@ RID RenderingDeviceVulkan::texture_create(const TextureFormat &p_format, const T
 		image_memory_barrier.sType = VK_STRUCTURE_TYPE_IMAGE_MEMORY_BARRIER;
 		image_memory_barrier.pNext = nullptr;
 		image_memory_barrier.srcAccessMask = 0;
-		image_memory_barrier.dstAccessMask = VK_ACCESS_SHADER_READ_BIT;
+		image_memory_barrier.dstAccessMask = VK_ACCESS_SHADER_READ_BIT | VK_ACCESS_SHADER_WRITE_BIT;
```

Comment:
I think we can fine grain a bit more here. SHADER_WRITE_BIT would make sense only of the storage usage bit is set.
If the texture has color buffer usage, then we can mask COLOR_ATTACHMENT_OUTPUT_BIT. Likewise, if it has depth stencil usage you can use EARLY_FRAGMENT_TESTS_BIT, and if it can be updated, then TRANSFER_BIT.

Reply:
I've removed the change and added instead a new commit with new logic for this barrier.